### ISSUE TITLE
IoT Core subscription expired

### ISSUE THREAD

**Issue by Roberto6969** [tuya/tuya-home-assistant#804]
Hello,

what to do when IoT Core subscription is expired but I don't want to pay to Tuya 25K or 50K for Corporate or Flagship edition?
Is there any acceptable way to make this integration alive again?

![image](https://user-images.githubusercontent.com/21200507/167306984-0591a498-e37f-4f1d-b5a4-e4e8904bd31d.png)

Re-subscribing to Free Edition (honestly, for one month only is not usable) is not possible

![image](https://user-images.githubusercontent.com/21200507/167307075-b4aa228f-39d8-4817-bc23-90dd2926c589.png)

Thank You.

**Comment by moyuehan** [comment]
There used to be an extension button, giving an extra 6 months IoT Core for free. It looks like the rules have changed again : I cannot find the extension button anymore. The only solution given me today by a Tuya "tech support" engineer is to do this every month renewal of the free resource pack for the IoT core service. Looks like this needs to be done AFTER the expiration date.

**Comment by Roberto6969** [comment]
It seems there is no reasonable solution for such a complicated system as Tuya have. I will quit with Tuya, there are much better and simpler solutions. Tuya failed with it's cloud totally. Even Sonoff is far better.

**Comment by moyuehan** [comment]
I contacted MarkWatttech to get possible new information since his excellent yt video on Tuya integration issues 9 months ago. He could not help this time. Why not call for help from Tuya ? At the Tuya  site https://iot.tuya.com/ Cloud Development Learn More Best Practice Integrating Home Assistant and Home Bridge How to set up Tuya integration in Home Assistant one can find answers to a lot of questions. Scroll down to : "Will I be billed after the free trial of the API service expires?". The answer is still 6 months extension possible. This is acceptable. 
However I could not find the button to request for this extension. Some 6 months ago it was there, and I used it succesfully. What happened ?
Then I contacted Tuya Tech Support - one of the headers in th Tuya IoT Platform and launched a written question, providing the link to the documentation. I immediately had support from a Tuya engineer. Now 2 days later I have the button for extension again, and I just launched the request for a 6 month extension.

**Comment by moyuehan** [comment]
In general, Tuya Tech Support was very helpfull. I had immediate response to my request. It clearly took some effort to reinstall the extension button again, with question for testing and trial. I was impressed by the speed of response, clear action by the Tuya tech support engineer. 
I will continue to trust Tuya and Tuya products. I have several smart plugs, IR controls, power monotoring via CT clamp, all working well in Home Assistant. The new Tuya V2 integration and HA works very well for me.

**Comment by VShade** [comment]
Bullshit Tuya is trash. Not useable for private home users.

**Comment by moyuehan** [comment]
After almost 2 years of use in my home automation I have positive
experience for devices and the Tuya IoT platform. As I am reading a lot of
negative comments, I checked the use of Shelly as backup solution...doesn't
work. Shelly does not cover the full range of devices Tuya offers. As for
the IoT Core, I can extend the free use every 6 months with another 6
months. I buy all devices at Aliexpress.

Op ma 31 okt. 2022 13:02 schreef VShade ***@***.***>:

**Comment by spuljko** [comment]
What are you complaining about? You bought cheap product which is not advertised as HA compatible and now you are mourning because you have to do extra few steps.

**Comment by VShade** [comment]
Extra fee steps? They chance the steps 2-3 time every year! I think, if they don't want end users, they should it post i front of her homepage ....

**Comment by zackforbing** [comment]
if anyone else encounters this, apparently it can be fixed from the Cloud Services tab:
![image](https://user-images.githubusercontent.com/13476440/208666471-2788660c-2eda-4a5f-b08c-ab085ed05b77.png)
![image](https://user-images.githubusercontent.com/13476440/208666528-44d96bc0-cf02-4844-b877-02a661a62010.png)

**Comment by moyuehan** [comment]
You take care of it every 6 months ?

Op di 20 dec. 2022 13:24 schreef Zack Forbing ***@***.***>:

**Comment by VShade** [comment]
I found a good way. Sell all tuya-devices and buy zigbee or tasmota....

**Comment by Roberto6969** [comment]
Definitely. Life with Tasmota is far better than Tuya. Tuya failed completely with it's platform. Even Sonoff is much better than Tuya. Advice for Tuya: don't complicate, don't broke subscription for devices  we didn't get for free.

**Comment by FilipeCastreze** [comment]
Any suggestions for light bulbs? I'm afraid I can't change the firmware to tasmota on these devices.

**Comment by ro8inmorgan** [comment]
I wanted to let everyone know there's another way to get all the local keys without the tuya cloud platform. Its a little bit more tricky but it does actually work. It involves running an older version of the smartlife app in either blue stacks or rooted android phone which actually saves all keys unencrypted to an xml file. Then its just a matter of reading this xml file to extract they keys. The maker of the video even provided a simple tool to extract the keys from the xml more simpler.

https://www.youtube.com/watch?v=YKvGYXw-_cE

**Comment by FilipeCastreze** [comment]
> I wanted to let everyone know there's another way to get all the local keys without the tuya cloud platform. Its a little bit more tricky but it does actually work. It involves running an older version of the smartlife app in either blue stacks or rooted android phone which actually saves all keys unencrypted to an xml file. Then its just a matter of reading this xml file to extract they keys. The maker of the video even provided a simple tool to extract the keys from the xml more simpler.
> 
> https://www.youtube.com/watch?v=YKvGYXw-_cE

Thank you  @ro8inmorgan

**Comment by Denys-Vasyliev** [comment]
Link to extension trial for 6 month

https://iot.tuya.com/cloud/products/apply-extension

**Comment by dJani97** [comment]
> Link to extension trial for 6 month
> 
> https://iot.tuya.com/cloud/products/apply-extension

I'm getting an error message when submitting this form: "All input parameters of the remote API are lost."
Looking at the Network tab, this is literally what the server responds with.

**Comment by el-lsan** [comment]
@dJani97 I had the same issue.
If you click on back button, and click on `Extend Trial Period` again it should work.

**Comment by AndySymons** [comment]
> if anyone else encounters this, apparently it can be fixed from the Cloud Services tab: ![image](https://user-images.githubusercontent.com/13476440/208666471-2788660c-2eda-4a5f-b08c-ab085ed05b77.png) ![image](https://user-images.githubusercontent.com/13476440/208666528-44d96bc0-cf02-4844-b877-02a661a62010.png)

Yep, this sill works at 29 June 2023: 
1. Log in to the Tuya IOT platform
2. Navigate Dev platform – Cloud – Cloud services
3. Find “IOT Core” in the list and click “view details”
4. It then gives you the current expiration date and a link to apply to extend it. The link opens a form you fill in with who you are and how many devices you have (less than 50, 50-1000 etc).
5. After sending the form the IOT core product has the status “Your application for extension is being reviewed” …

**Comment by zomar** [comment]
> > if anyone else encounters this, apparently it can be fixed from the Cloud Services tab: ![image](https://user-images.githubusercontent.com/13476440/208666471-2788660c-2eda-4a5f-b08c-ab085ed05b77.png) ![image](https://user-images.githubusercontent.com/13476440/208666528-44d96bc0-cf02-4844-b877-02a661a62010.png)
> 
> Yep, this sill works at 29 June 2023:
> 
> 1. Log in to the Tuya IOT platform
> 2. Navigate Dev platform – Cloud – Cloud services
> 3. Find “IOT Core” in the list and click “view details”
> 4. It then gives you the current expiration date and a link to apply to extend it. The link opens a form you fill in with who you are and how many devices you have (less than 50, 50-1000 etc).
> 5. After sending the form the IOT core product has the status “Your application for extension is being reviewed” …

Super helpful, works at 1 September 2023, thank you.

**Comment by moyuehan** [comment]
It is an extension for 6 months. Please note the day it will expire. A day
earlier, look again for a next extension. I commented to Tuya for longer
extensions without a positive answer. Anyhow I am getting used to the
extension request every 6 months.

Op za 2 sep. 2023 07:28 schreef zomar ***@***.***>:

**Comment by AndySymons** [comment]
I found this too much and went back to using cloud Tuya

**Comment by moyuehan** [comment]
You now use Cloud Tuya only, or still in combination with HA ?

Op za 9 sep. 2023 14:39 schreef Andy Symons ***@***.***>:

**Comment by Metroseksuaali** [comment]
I wish i would have found these problems before i bought my lights. Well only 4 of them but im not going to pay 25000€/ Year for 4 lightbulbs :D After 6 months i will just throw them to trash

**Comment by moyuehan** [comment]
You can continue to use them for years in Tuya Smart Life as well as in
Home Assistant. In Google Home or Amazon Alexa everything continues to work
fine. For Home Assistant you need to extend the development status. It is
not costing anything. I am doing it already for 2 years now. I will make
some screen shots to make it easy for you.

Op di 17 okt. 2023 14:11 schreef Metroseksuaali ***@***.***>:

**Comment by Floriszz** [comment]
Stupid, I went the same route and ended up here, although I was following these instructions below, I fell in the 'trap' of 'Tuya onboarding' to activate the IOT core trial automatically by creating my account.
Instructions using the "basic edition" for "Smart Home Scene Linkage";
"Smart Home Scene Linkage"

I haven't re-tried jet, but the 'crucks' is that you do NOT activate the initial IOT core trial, but the "Smart Home Scene Linkage"(https://www.tuya.com/vas/commodity/SMART_HOME_SCENE_MANAGEMENT_V2), which is now stated 'Deprecated' but is free for ''Long-Lived"(for whatever that means).
![image](https://github.com/user-attachments/assets/47f92a61-02dc-4c4c-b13d-b59a923a570b)
If I try to subscribe to it now, while having the IOT core trial active as well, it says I can only use 1 trial....
![image](https://github.com/user-attachments/assets/37a072e6-88c7-4e89-b55e-1fec97f1eb6e)

![image](https://github.com/user-attachments/assets/547ddfde-5dd5-431e-a292-19ef200bc0f1)

**Comment by moyuehan** [comment]
I think it is not to bad to go once every 6 months to subscribe again.
Keeps me familiar with Tuya website, and its changes. I need to go there
anyhow for local keys to use for 'local Tuya'

Op vr 12 jul. 2024 09:48 schreef Florisz ***@***.***>:

**Comment by thiagosanches** [comment]
Well, I just asked for a renewal and it seems that worked, I'll test it soon.
![image](https://github.com/user-attachments/assets/8e9d2e0e-1a03-4bcb-8456-c13d7cc9705f)

That suggestion that someone gave us, to click on the back button to avoid that message "All input parameters of the remote API are lost." did the trick.

> @dJani97 I had the same issue. If you click on back button, and click on `Extend Trial Period` again it should work.

**Comment by moyuehan** [comment]
Yes. I also keep using the renewal. Very usefull for getting 'local key'
and an excellent overview of my almost 50 Tuya devices.

Op vr 9 aug. 2024 16:12 schreef Thiago Sanches ***@***.***>:

**Comment by IIIdefconIII** [comment]
![Image](https://github.com/user-attachments/assets/e8a0162c-ee48-46e8-acfd-6c203e0f5026)

**Comment by JakubSs** [comment]
> if anyone else encounters this, apparently it can be fixed from the Cloud Services tab: ![image](https://user-images.githubusercontent.com/13476440/208666471-2788660c-2eda-4a5f-b08c-ab085ed05b77.png) ![image](https://user-images.githubusercontent.com/13476440/208666528-44d96bc0-cf02-4844-b877-02a661a62010.png)

Thank you, still working like a charm

**Comment by moyuehan** [comment]
Yes, still works...one difficulty only...need to renew every 6 months. On
the other hand, Tuya's developpers website looks great and has a lot of
advantages.

Op wo 2 jul. 2025 22:50 schreef Jakub ***@***.***>:

**Comment by albeedelacruz** [comment]
Is there an easy way to auto-renew? I'm thinking via Chrome-Selenium to automate via script but a lot of things could change every 6 months so by the time the script runs, parameters may no longer be applicable.

**Comment by moyuehan** [comment]
The easiest way to keep the connection Tuya/Home Assistant is the 6 monthly
renewal of IoT Core. I am doing it for 5 years now. To make it easy for
myself, I put it as an appointment in my Google Agenda, with a reminder
every 6 months, for the next 10 years. In the appointment description I
added all the steps to fill out the extension. For contact information
write your email address. The day it expires, you get a notification via
this email address.
There is another more complex way : install via HACS the custom integration
'Tuya Local' (https:github.com/make-all/tuya-local). It applies to all yr
existing Tuya devices. Pay attention to use the correct repository. Now you
are free from the 6 monthly subscription.

Op do 14 mei 2026 09:51 schreef albeedelacruz ***@***.***>:

**Comment by albeedelacruz** [comment]
> The easiest way to keep the connection Tuya/Home Assistant is the 6 monthly
> renewal of IoT Core. I am doing it for 5 years now. To make it easy for
> myself, I put it as an appointment in my Google Agenda, with a reminder
> every 6 months, for the next 10 years. In the appointment description I
> added all the steps to fill out the extension. For contact information
> write your email address. The day it expires, you get a notification via
> this email address.
> There is another more complex way : install via HACS the custom integration
> 'Tuya Local' (https:github.com/make-all/tuya-local). It applies to all yr
> existing Tuya devices. Pay attention to use the correct repository. Now you
> are free from the 6 monthly subscription.
> 
> Op do 14 mei 2026 09:51 schreef albeedelacruz ***@***.***>:
> […](#)

I appreciate your response. I’ll try to automate this and see how it goes over the next 6 months. 🙂

Yes, I’ve looked into LocalTuya, but it seems it wouldn’t work for my particular use case. Unfortunately, I’m using some inexpensive Zigbee locks that apparently only work with Tuya rather than through ZHA, and the unlocking feature also isn’t supported by the LocalTuya integration.

Because of that, I didn’t really have another option besides using the Tuya Smart Lock integration, which relies on the IoT Core and Smart Lock API nodes.

**Comment by moyuehan** [comment]
Maybe have a look at Tuya Quirks for the zigbee devices
https://github.com/zigpy/zha-device-handlers/blob/dev/tuya.md

Op do 14 mei 2026 16:40 schreef albeedelacruz ***@***.***>:
